src/shared/copy.c

```diff
@@ -952,33 +953,24 @@ static int fd_copy_directory(
         if (!d)
                 return -errno;
 
-        exists = false;
-        if (copy_flags & COPY_MERGE_EMPTY) {
-                r = dir_is_empty_at(dt, to, /* ignore_hidden_or_backup= */ false);
-                if (r < 0 && r != -ENOENT)
-                        return r;
-                else if (r == 1)
-                        exists = true;
-        }
+        r = dir_is_empty_at(dt, to, /* ignore_hidden_or_backup= */ false);
+        if (r > 0 && !(copy_flags & (COPY_MERGE|COPY_MERGE_EMPTY)))
+                return -EEXIST;
+        else if (r == 0 && !FLAGS_SET(copy_flags, COPY_MERGE))
+                return -EEXIST;
+        else if (r < 0 && r != -ENOENT)
+                return r;
```

Comment:
the word `else` is redudnat each case.

the first two branches are the same?

Reply:
They're not? If the dir is empty, we want to consider both `COPY_MERGE` and `COPY_MERGE_EMPTY`. If the dir is not empty, we only want to consider `COPY_MERGE`.

Reply:
Ah sure, I just figured it'd be more readable like this, will merge